### ISSUE TITLE
Options covered by TranslationRequest

### ISSUE THREAD

**Issue by jerinphilip** [browsermt/bergamot-translator#89]
Meta issue to discuss and complete docs for keys and possible values for the message passed in regarding what or how should Response be constructed.

@abhi-agg 
> We will have to add a documentation listing all the keys and the corresponding values that can be provided as translation request.

@motin This is where I want your inputs, this is not API design, this is slight change/discussion in what you communicate to me and what I respond with. Unified API is a wall which changes the objectives to something else and an unnecessary time-sink. I put forth the following configurable parameters.
 
```yaml
alignment: true # true | false
alignment-threshold: 0.2f # Float value
quality: false # true | false
quality-score-type: free # free | expensive
concat-strategy: faithful # faithful | space 
```

<details> <summary> Explanation </summary>

1. `alignment-threshold`: So alignments is a (dense) matrix per [Unified API Example](https://github.com/mozilla/bergamot-translator/wiki/Unified-API-Example#cases). This is wasteful, as the matrix is often sparse and your algorithm is expected to only operate with what is the high-match alignments. I'd therefore like to provide you this additional configurability as well, where you set this to `0.0f` where you need the full alignment (the dense matrix) or some other tuned value where you want to experiment with different configurations. 
2. `quality-score-type`: I can offer you a free quality score as of now, which should help you develop UI components. However, I cannot guarantee the API remains same as we accommodate both Mozilla and Sheffield requirements. We're effectively parallelizing development with a bit of overhead here. I have some background developing UIs and particularly with quality scores and I'll add [this](https://jerinphilip.github.io/d3-sandbox/exps/uc-viz/) here to establish the credentials. You should be able to reuse UI components and run a few iterations while we make slight tweaks in the backend to get different but close to these structures quality.
3. `concat-strategy`: I am not sure if you want to have this, but you might already be aware that there are newline no newline etc issues with bergamot-translator. You can ask me here to translate text faithful to it's source structure or not if such provisions are present. Think you're translating a `.txt`, you can offload everything down and print back what we provide - in which case you'd want faithful. Not so much so if you're working with sentences picked up from HTML nodes.
</details>

We can add many more as we go ahead. With a dict, the possibilities increase. We'll also need some place to document these, maybe the wiki here or sphinx being generated. Let know your suggestions, or maybe more configurability you want.

Edit: Added quality score yes/no option.

**Comment by motin** [comment]
@jerinphilip 
1. I am on board with the ability to configure an alignment threshold on a per-request basis. Not having to traverse the alignments matrix makes the integration easier as a consumer of the API.
2. I am on board with the ability to configure which type of quality score that should be returned (as well as configuring not to return any quality scores at all for the particular request).
3. The text picked up from HTML nodes can be paragraphs of text with several newlines at various points. As a consumer, I would expect these to be preserved by the API, so no need for a configuration parameter here. Stay faithful all the time. :)

**Comment by jerinphilip** [comment]
I added `quality = true | false`. Most of the returned stuff will be made optional according to provided parameters. 

@abhi-agg I've started a page for comprehensive documentation of options at:

* https://github.com/browsermt/bergamot-translator/wiki/Options-covered-by-TranslationRequest

Guessing it's best to move into a `.md` in source to be picked up by doc tooling after we reach a consensus on the parameters and the documentation and it's implemented in source, until which all of us can enjoy fast and WYSIWYG editing.
